libmscore/repeatlist.cpp

```diff
@@ -764,7 +769,7 @@ void RepeatList::unwind()
     std::pair<QList<QList<RepeatListElement*>*>::const_iterator, QList<RepeatListElement*>::const_iterator> continueAt = std::make_pair(
         _rlElements.cend(), _rlElements[0]->cend());
     Jump const* activeJump = nullptr;
-    bool forceFinalRepeat = false;   // Used during jump processing
+    bool forceFinalRepeat;   // Used during jump processing
```

Comment:
better to save first initialization

Reply:
Why? It only slows down.

Reply:
OK, I see now it is part of the guidelines.

At least I'd rather keep initialization to an unused value here instead of moving declaration inside the loop which would be even more wasted instructions.

Reply:
Force pushed it

Reply:
I'll just explain why this is so :)
it is an undefined state. After this line it is not clear whether it is true or false.
It is even worse if someone decides to use this flag before initialization. Just safer, it's hard to reproduce bugs like this :)

Reply:
Yes, and usually I'd agree that initialization is a must from a safety pov. in this situation however, the comment clarifies the intended use of the variable, which "logically" should only be declared/scoped within the loop.
The only reason for not doing so, is to prevent continuous stack allocations which would slow down processing.

Similarly, you now take a penalty for explicitly initializing something to a value that is irrelevant (and should be at that stage). If at some future point in time someone decides indeed to use that flag, it would be part of their modification and responsibility to ensure it is initialized correctly for their use case.


I know that the above statement perhaps comes across a bit harsh; but I am misformed by years of embedded safety programming, where there are quite some rules about when initialization is required (indeed, as a ground rule, always) and when it should be omitted as part of the intent of the declaration.

With regards to this PR specifically, it was a mere port of the one for Mu3, so I try to keep most things as similar as possible. I'll gladly perform a code review session once 3.x is over and out (somewhere after 3.6.1 likely) to bring the whole file up to Mu4 standards and guidelines.

Reply:
(as an aside, note the similar approach for the playCount variable just above it; we probably should make that follow the rules in a future update then as well ;-) )